test/test_priority.py

```diff
@@ -143,6 +143,37 @@ def test_stream_ordering(self, a, b):
         assert (s1 == s2) == (a == b)
         assert (s1 != s2) == (a != b)
 
+    @pytest.mark.parametrize('bad_weight', [
+        None,
+        0.5,
+        float('inf'),
+        'priority',
+        object
+    ])
+    def test_non_integer_weight_is_error(self, bad_weight):
+        """
+        Creating a stream with a non-integer weight is rejected.
+        """
+        with pytest.raises(priority.BadWeightError) as err:
+            priority.Stream(stream_id=0, weight=bad_weight)
+        assert err.value.args[0] == 'Stream weight should be an integer'
```

Comment:
Rather than test these by constructing the stream object directly, it might be better to use the public API methods for this test. That way, if we ever move this validation the tests will continue to reflect the actual intent.